src/comics/chapters/chapter_comments/routes.rs

```diff
@@ -0,0 +1,211 @@
+        (status = StatusCode::INTERNAL_SERVER_ERROR, description = "Something went wrong", body = ErrorResponse),
+    ),
+    tag = "Chapter Comments API"
+)]
+#[axum::debug_handler(state = AppState)]
+pub async fn create_comment(
+    auth: AuthExtractor<{ UserRole::User as u32 }>,
+    State(state): State<Arc<InnerAppState>>,
+    Path(chapter_id): Path<Uuid>,
+    Json(payload): Json<CreateChapterComment>,
+) -> Result<Json<ChapterCommentResponse>, ChapterCommentsError> {
+    let mut db = state.pool.get().await?;
+
+    let comment = db
+        .transaction::<_, ChapterCommentsError, _>(|transaction| {
+            async move {
+                let comment = ChapterComment {
+                    id: Uuid::now_v7(),
+                    content: payload.content,
+                    created_at: Utc::now(),
+                    updated_at: None,
+                    chapter_id,
+                    user_id: auth.current_user.id,
+                };
+
+                let comment = diesel::insert_into(chapter_comments::table)
+                    .values(&comment)
+                    .get_result::<ChapterComment>(transaction)
+                    .await?;
+
+                let commnet_response = ChapterCommentResponse {
+                    id: comment.id,
+                    chapter_id,
+                    content: comment.content,
+                    user: auth.current_user,
+                    parent_comment: payload.parent_comment_id,
+                    child_comments_ids: vec![],
+                    child_comments: vec![],
+                };
+
+                if let Some(parent_comment_id) = payload.parent_comment_id {
+                    diesel::insert_into(chapter_comments_mapping::table)
+                        .values((
+                            chapter_comments_mapping::parent_comment_id.eq(parent_comment_id),
+                            chapter_comments_mapping::child_comment_id.eq(comment.id),
+                        ))
+                        .execute(transaction)
+                        .await?;
+                }
+
+                Ok(commnet_response)
+            }
+            .scope_boxed()
+        })
+        .await?;
+
+    Ok(Json(comment))
+}
+
+#[utoipa::path(
+    delete,
+    path = "/api/v1/comics/chapters/:chapter_id/comments",
+    responses(
+        (status = 200, description = "Specified comment has been successfully deleted. returned deleted comment's ID", body = Uuid),
+        (status = StatusCode::BAD_REQUEST, description = "Invalid Comment ID", body = ErrorResponse),
+        (status = StatusCode::INTERNAL_SERVER_ERROR, description = "Something went wrong", body = ErrorResponse),
+    ),
+    tag = "Chapter Comments API"
+)]
+#[axum::debug_handler(state = AppState)]
+pub async fn delete_comment(
+    auth: AuthExtractor<{ UserRole::User as u32 }>,
+    State(state): State<Arc<InnerAppState>>,
+    Path(comment_id): Path<Uuid>,
```

Comment:
this is picking the chapter id.
the route should be `/api/v1/comics/chapters/comics/:comment:id`

Reply:
done (it's `/api/v1/comics/chapters/comments/comment_id` btw)